pyiceberg/table/puffin.py

```diff
@@ -114,3 +147,97 @@ def __init__(self, puffin: bytes) -> None:
 
     def to_vector(self) -> dict[str, "pa.ChunkedArray"]:
         return {path: _bitmaps_to_chunked_array(bitmaps) for path, bitmaps in self._deletion_vectors.items()}
+
+
+class PuffinWriter:
```

Comment:
Iceberg spec recommends writing `created-by` property in Puffin files: https://iceberg.apache.org/puffin-spec/#deletion-vector-v1-blob-type. Could you please write the property?